### PR TITLE
Refactoring: Renaming types, files, splitting files.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Review by mavasani — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@333fred, @cston, @dotnet/roslyn-compiler Please review

**Review by cston — APPROVED** [review]

File: src/Compilers/Core/Portable/Operations/BasicBlock.cs [hunk 1]
```diff
@@ -1,121 +1,113 @@
...
+        }
 
-            public IReadOnlyCollection<IOperation> Statements => _statementsBuilder;
-            public ImmutableArray<IOperation> GetImmutableStatements() => _statementsBuilder.ToImmutableAndFree();
-            public ISet<BasicBlock> Predecessors => _predecessorsBuilder;
-            public BasicBlockKind Kind { get; }
+        public BasicBlockKind Kind { get; }
 
-            public int Ordinal { get; set; } = -1;
+        public ImmutableArray<IOperation> Operations { get; }
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Operations/BasicBlock.cs]: Nit: consider moving the auto-properties up (together with the fields) so it's easier to understand what the block stores.

**Review by jcouv — APPROVED** [review]
LGTM

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.BasicBlockBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            public bool IsReachable { get; set; } = false;
+
+            public ControlFlowRegion Region { get; set; }
+
+            internal void AddStatement(IOperation statement)
+            {
+                _statementsBuilder.Add(statement);
+            }
+
+            internal void AddStatements(IEnumerable<IOperation> statements)
```
Comment (333fred) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.BasicBlockBuilder.cs]: @AlekseyTs were you planning on adjusting this in another followup?

Comment (333fred) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.BasicBlockBuilder.cs]: Ah, I see that you are.

---
In reply to: [192534549](https://github.com/dotnet/roslyn/pull/27315#discussion_r192534549) [](ancestors = 192534549)

**Pull request merged**